lib/react_client.dart

```diff
@@ -36,19 +36,92 @@ newJsMap(Map map) {
 }
 
 /**
- * Type of [children] must be child or list of childs, when child is JsObject or String
+ * Type of [children] must be child or list of children, when child is JsObject or String
  */
 typedef JsObject ReactComponentFactory(Map props, [dynamic children]);
 typedef Component ComponentFactory();
 
 class ReactComponentFactoryProxy implements Function {
-  final ReactComponentFactory _call;
   final JsFunction reactComponentFactory;
-  ReactComponentFactoryProxy(this.reactComponentFactory, this._call);
+  ReactComponentFactoryProxy(this.reactComponentFactory);
 
   JsObject call(Map props, [dynamic children]) {
-    return this._call(props, children);
+    List<dynamic> reactParams = [_generateExtendedJSProps(props, children)];
+    reactParams.add(children);
```

Comment:
Looks like it does. When `children` of type `List` are passed to React as a non-`JsArray`, React's internals only sees them as a `DartObject`, so it cannot validate them properly. It may also have other negative effects on how React handles `children` internally.

I added test coverage and a fix in this PR: https://github.com/trentgrover-wf/react-dart/pull/1

Revert the changes in `react_client.dart` and rerun `child_key_warning_test`, and you'll see that validation doesn't work properly when you pass children as a `List` to a Dart component.